### PR TITLE
basic float printing

### PR DESCRIPTION
**Pull request by MovingtoMars**
This PR adds support for basic printing of f64. Note that the output is rather human-unfriendly. For example, `500.0` is printed as `8796093022208000*2^-44`.

### CODE REVIEW

File: std/std.zig [hunk 1]
```diff
@@ -229,6 +243,68 @@ fn buf_print_u64(out_buf: []u8, x: u64) -> isize {
     return len;
 }
 
+pub fn buf_print_f64(out_buf: []u8, x: f64) -> isize {
+    if (x == f64_get_pos_inf()) {
+        const buf2 = "+Inf";
+        @memcpy(&out_buf[0], &buf2[0], 4);
```
Comment (andrewrk) [inline std/std.zig]: instead of 4, `buf2.len`

File: std/std.zig [hunk 2]
```diff
@@ -229,6 +243,68 @@ fn buf_print_u64(out_buf: []u8, x: u64) -> isize {
...
+
+    var len: isize = 0;
+
+    // 1 sign bit
+    // 11 exponent bits
+    // 52 significand bits (+ 1 implicit always non-zero bit)
+
+    const bits = f64_to_bits(x);
+    if (bits & (1 << 63) != 0) {
+        %%stdout.printf("neg\n");
```
Comment (andrewrk) [inline std/std.zig]: it's not supposed to print to stdout right?

File: std/std.zig [hunk 3]
```diff
@@ -229,6 +243,68 @@ fn buf_print_u64(out_buf: []u8, x: u64) -> isize {
...
+        buf[0] = '-';
+        len += 1;
+    }
+
+    const rexponent: i64 = i64((bits >> 52) & ((1 << 11) - 1));
+    const exponent = rexponent - 1023 - 52;
+    /*%%stdout.printf("exponent: ");
```
Comment (andrewrk) [inline std/std.zig]: this code looks like WIP, do you want it merged now?

File: std/std.zig (hunk 2 above)
Comment (MovingtoMars) [inline std/std.zig]: Oh, forgot to remove that.

File: std/std.zig (hunk 3 above)
Comment (MovingtoMars) [inline std/std.zig]: Depends on whether you want basic float printing right now or not.

Comment (andrewrk) [inline std/std.zig]: hmm, well it's better than status quo, so let's do it.

**Pull request merged**